### PR TITLE
Add new Cryptocurrency

### PR DESCRIPTION
**Pull request by nicosanchezprev**
**3rd task: Add new Cryptocurrency**

> Things the user can do:

- Add a new cryptocurrency from the api with the name or the symbol.
- Delete the cryptos from the list.
- Refresh the app and keep he's cryptos added.
- There are a error controller when adding a new crypto.

### CODE REVIEW

File: components/ModalInput/index.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    borderPX: '1px',
+  };
+  const themePresTextChange = {
+    col: theme.colors.blue,
+  };
+  const themePresTextDefault = {
+    col: theme.colors.grey2,
+  };
+
+  const addCryptocurrencie = () => {
```
Comment (VicenBayugar) [inline components/ModalInput/index.tsx]: cryptocurrency

File: components/ModalInput/index.tsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+              value={inputValue}
+              placeholder="Use a name or ticker symbol..."
+              keyboardType="default"
+              onFocus={() => setIsActive(true)}
+              onBlur={() => setIsActive(false)}
+            />
+          </ThemeProvider>
+          <PressableAdd
+            onPress={() => {
+              addCryptocurrencie();
```
Comment (VicenBayugar) [inline components/ModalInput/index.tsx]: onPress={addCryptocurrencie}

File: redux/reducersComp/cryptosSlice.ts [hunk 3]
```diff
@@ -35,14 +37,39 @@ export const cryptoApiData = createAsyncThunk(
...
+      state.cryptosData = state.cryptosData.filter(
+        crypto => crypto.name !== action.payload,
+      );
+      removeValue(action.payload);
+    },
+  },
   extraReducers: builder => {
     builder.addCase(cryptoApiData.fulfilled, (state, action) => {
-      const check = state.cryptosData.find(
+      let check = state.cryptosData.find(
```
Comment (VicenBayugar) [inline redux/reducersComp/cryptosSlice.ts]: why did you change it to let?

File: utils/asyncFunctions.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    await AsyncStorage.setItem(`${value.name}`, jsonValue);
+  } catch (e: any) {
+    console.log(e);
+  }
+};
+
+export const removeValue = async (keyName: string) => {
+  try {
+    await AsyncStorage.removeItem(keyName);
+    console.log('Removed from async-storage!');
```
Comment (VicenBayugar) [inline utils/asyncFunctions.ts]: delete this log

File: utils/asyncFunctions.ts [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  } catch (e: any) {
+    console.log(e);
+  }
+};
+
+export const getMultiple = async () => {
+  try {
+    let values;
+    let asyncKeys = await getAllKeys();
+    console.log('asynckeys', asyncKeys);
```
Comment (VicenBayugar) [inline utils/asyncFunctions.ts]: delete this log

File: components/ListCripto/index.tsx [hunk 6]
```diff
@@ -25,6 +37,7 @@ const ListCripto = ({setModal}: ListCriptoProps) => {
           </PressableAdd>
         }
       />
+      {error !== '' ? errorAlert() : null}
```
Comment (tomasdematos) [inline components/ListCripto/index.tsx]: this should be inside a useEffect, and if the error updates, then show the alert

File: components/ModalInput/index.tsx [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+            <Input
+              onChangeText={setInputValue}
+              value={inputValue}
+              placeholder="Use a name or ticker symbol..."
+              keyboardType="default"
+              onFocus={() => setIsActive(true)}
+              onBlur={() => setIsActive(false)}
+            />
+          </ThemeProvider>
+          <PressableAdd
```
Comment (tomasdematos) [inline components/ModalInput/index.tsx:46]: this button needs validation to be disabled

File: redux/reducersComp/cryptosSlice.ts (hunk 3 above)
Comment (nicosanchezprev) [reply redux/reducersComp/cryptosSlice.ts]: My mistake, when i did the PR, there was some problems and i did some mixes in the code. I will let it as const again.

File: components/Criptotracker/index.tsx [hunk 8]
```diff
@@ -1,22 +1,28 @@
...
   const [modal, setModal] = useState(false);
   const dispatch = useAppDispatch();
-
   useEffect(() => {
-    dispatch(cryptoApiData('btc'));
-    dispatch(cryptoApiData('usdt'));
-    dispatch(cryptoApiData('eth'));
+    const asyncSearch = async () => {
+      const allStorage = await getMultiple();
+      allStorage?.map(item => {
```
Comment (FedericoLeiva12) [inline components/Criptotracker/index.tsx]: It should be a forEach, you aren't using the result of the map

File: components/ListCripto/index.tsx [hunk 9]
```diff
@@ -1,12 +1,27 @@
...
+  useEffect(() => {
+    const errorAlert = () => {
+      Alert.alert('Error', `${error}`, [
+        {
+          text: 'OK',
+          onPress: () => dispatch(cleanError()),
+        },
+      ]);
+    };
+    error !== '' ? errorAlert() : null;
```
Comment (tomasdematos) [inline components/ListCripto/index.tsx]: Use an if instead of a ternary

File: components/Criptotracker/index.tsx [hunk 10]
```diff
@@ -1,22 +1,28 @@
...
+        dispatch(addCrypto(item[1] ? JSON.parse(item[1]) : null));
+      });
+    };
+    asyncSearch();
   }, [dispatch]);
 
   return (
     <PrincipalView>
```
Comment (FedericoLeiva12) [inline components/Criptotracker/index.tsx]: You should use Main as the name

File: components/ListCripto/index.tsx [hunk 11]
```diff
@@ -1,12 +1,27 @@
...
 import {Nav, PressableAdd, PressableAddText, Title} from './styles';
-import {useAppSelector} from '../../redux/hooks/hooks';
+import {useAppDispatch, useAppSelector} from '../../redux/hooks/hooks';
+
 import ItemCripto from '../ItemCripto';
+import {cleanError} from '../../redux/reducersComp/cryptosSlice';
 
 const ListCripto = ({setModal}: ListCriptoProps) => {
-  const {cryptosData} = useAppSelector(state => state.crypto);
+  const {cryptosData, error} = useAppSelector(state => state.crypto);
```
Comment (FedericoLeiva12) [inline components/ListCripto/index.tsx]: use destructuring

File: components/ModalInput/index.tsx [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+  const dispatch = useAppDispatch();
+
+  const addCryptocurrency = () => {
+    dispatch(cryptoApiData(inputValue)).then(() => {
+      setModal(false);
+    });
+  };
+
+  return (
+    <Modal visible={modal} animationType={'slide'}>
```
Comment (tomasdematos) [inline components/ModalInput/index.tsx]: use  animationType="slide"

File: components/ModalInput/index.tsx [hunk 13]
```diff
@@ -0,0 +1,57 @@
...
+            onChangeText={setInputValue}
+            value={inputValue}
+            placeholder="Use a name or ticker symbol..."
+            keyboardType="default"
+            onFocus={() => setIsActive(true)}
+            onBlur={() => setIsActive(false)}
+            col={isActive}
+          />
+          <PressableAdd
+            disabled={inputValue === '' ? true : false}
```
Comment (tomasdematos) [inline components/ModalInput/index.tsx]: use only inputValue === ''", it is already a boolean

File: utils/asyncFunctions.ts [hunk 14]
```diff
@@ -0,0 +1,44 @@
...
+  } catch (e: any) {
+    console.log(e);
+  }
+};
+
+export const removeValue = async (keyName: string) => {
+  try {
+    await AsyncStorage.removeItem(keyName);
+  } catch (e: any) {
+    console.log(e);
```
Comment (FedericoLeiva12) [inline utils/asyncFunctions.ts]: Change it to console.error

File: utils/asyncFunctions.ts [hunk 15]
```diff
@@ -0,0 +1,44 @@
...
+
+export const getAllKeys = async () => {
+  let keys;
+  try {
+    keys = await AsyncStorage.getAllKeys();
+    if (keys !== undefined) {
+      return keys;
+    }
+  } catch (e: any) {
+    console.log(e);
```
Comment (FedericoLeiva12) [inline utils/asyncFunctions.ts]: Change it to console.error

File: utils/asyncFunctions.ts [hunk 16]
```diff
@@ -0,0 +1,44 @@
...
+
+export const getMultiple = async () => {
+  try {
+    let values;
+    let asyncKeys = await getAllKeys();
+    console.log('asynckeys', asyncKeys);
+    values = await AsyncStorage.multiGet(asyncKeys ? asyncKeys : ['null']);
+    return values;
+  } catch (e: any) {
+    console.log(e);
```
Comment (FedericoLeiva12) [inline utils/asyncFunctions.ts]: Change it to console.error

File: utils/asyncFunctions.ts (hunk 5 above)
Comment (FedericoLeiva12) [reply utils/asyncFunctions.ts]: This is not solved

File: components/ModalInput/styles.ts [hunk 17]
```diff
@@ -0,0 +1,62 @@
...
+  color: ${theme.colors.black};
+`;
+
+interface PropsInput {
+  col: boolean;
+}
+
+export const Input = styled.TextInput<PropsInput>`
+  background-color: ${props =>
+    props.col === true ? theme.colors.white : theme.colors.skyBlue};
```
Comment (tomasdematos) [inline components/ModalInput/styles.ts]: props.col === true, if col is already a boolean, there is no need of validation

if prop.col === true, return true, so it's the same to put only prop.col
the same if it's false

File: components/ModalInput/index.tsx (hunk 13 above)
Comment (FedericoLeiva12) [inline components/ModalInput/index.tsx]: Change it to `inputValue !== ''` or something like that

**Review by FedericoLeiva12 — CHANGES_REQUESTED** [review]
